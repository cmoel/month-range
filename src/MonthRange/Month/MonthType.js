import * as R from "ramda";
import daggy from "daggy";

const isNil = R.isNil;
const lt = R.flip(R.lt);
const gt = R.flip(R.gt);
const lte = R.flip(R.lte);
const gte = R.flip(R.gte);

const Month = daggy.tagged("Month", ["year", "position"]);

Month.from = R.compose(R.apply(Month), R.props(["year", "position"]));

Month.equals = R.curry(
  (m1, m2) => R.equals(m1.year, m2.year) && R.equals(m1.position, m2.position)
);

Month.lte = R.curry((m1, m2) => {
  if (lt(m1.year, m2.year)) return true;
  return lte(m1.year, m2.year) && lte(m1.position, m2.position);
});

Month.gte = R.curry((m1, m2) => {
  if (gt(m1.year, m2.year)) return true;
  return gte(m1.year, m2.year) && gte(m1.position, m2.position);
});

Month.isBetween = R.curry((first, last, month) =>
  R.cond([
    [R.always(isNil(first)), R.F],
    [R.always(isNil(last)), Month.equals(first)],
    [R.T, R.both(Month.gte(first), Month.lte(last))],
  ])(month)
);

Month.isIncludedIn = R.curry(({fst, snd}, month) =>
  R.compose(
    R.apply(R.__, [month]),
    R.partial(Month.isBetween),
    R.sortWith([R.ascend(R.prop("year")), R.ascend(R.prop("position"))])
  )([fst, snd])
);

export default Month;
