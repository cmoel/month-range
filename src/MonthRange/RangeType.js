import * as R from "ramda";
import daggy from "daggy";

const isNil = R.isNil;
const notNil = R.complement(R.isNil);

const Term = daggy.tagged("Term", ["fst", "snd"]);

Term.empty = () => Term(undefined, undefined);

Term.from = R.compose(R.apply(Term), R.props(["fst", "snd"]));

Term.fstSndNil = ({fst, snd}) => R.and(isNil(fst), isNil(snd));
Term.fstSndSelected = ({fst, snd}) => R.and(notNil(fst), notNil(snd));
Term.fstSelected = ({fst, snd}) => R.and(notNil(fst), isNil(snd));
Term.fstReselected = R.curry((month, {fst, snd}) =>
  R.and(R.equals(fst, month), isNil(snd))
);

Term.setFst = R.curry((month, _term) => Term(month, undefined));
Term.setSnd = R.curry((month, {fst}) => Term(fst, month));

Term.select = R.curry((month, term) => {
  return R.cond([
    [Term.fstSndNil, Term.setFst(month)],
    [Term.fstSndSelected, Term.setFst(month)],
    [Term.fstReselected(month), Term.setFst(undefined)],
    [Term.fstSelected, Term.setSnd(month)],
  ])(term);
});

export default Term;
