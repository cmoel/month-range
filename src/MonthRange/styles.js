import glamorous from "glamorous";

const {div, header, h1} = glamorous;

export const Container = div({
  margin: "42px auto 0",
  maxWidth: 320,
});

export const Header = header({
  alignContent: "center",
  alignItems: "center",
  display: "flex",
  flexDirection: "row",
  justifyContent: "space-between",
});

export const H1 = h1({
  color: "#333",
  fontSize: "1em",
  margin: 0,
  letterSpacing: "0.10em",
  textTransform: "uppercase",
});
