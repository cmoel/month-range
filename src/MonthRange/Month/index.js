import React, {Component} from "react";
import * as R from "ramda";

import {Div} from "./styles";

import MonthType from "./MonthType";

const classNameFor = ({fst, snd}, month) =>
  R.cond([
    [R.equals(fst), R.always("fst")],
    [R.equals(snd), R.always("snd")],
    [MonthType.isIncludedIn({fst, snd}), R.always("selected")],
    [R.T, R.always("")],
  ])(month);

class Month extends Component {
  render() {
    const {year, position, selectMonth, selection, children} = this.props;
    const month = MonthType.from({year, position});
    return (
      <Div
        className={classNameFor(selection, month)}
        onClick={() => selectMonth(month)}
        position={position}
      >
        {children}
      </Div>
    );
  }
}

export default Month;
