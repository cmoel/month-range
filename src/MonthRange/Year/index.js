import React, {Component} from "react";

import {Months, YearDiv} from "./styles";

class Year extends Component {
  render() {
    const {year, selectMonth, selection, children} = this.props;
    return (
      <article>
        <YearDiv>{year}</YearDiv>
        <Months>
          {React.Children.map(children, c =>
            React.cloneElement(c, {year, selection, selectMonth})
          )}
        </Months>
      </article>
    );
  }
}

export default Year;
