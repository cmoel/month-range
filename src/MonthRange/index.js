import React, {Component} from "react";

import {Container, Header, H1} from "./styles";

import RangeType from "./RangeType";

import Year from "./Year";
import Month from "./Month";

class MonthRange extends Component {
  constructor() {
    super();
    this.state = RangeType.empty();
  }

  selectMonth = month => {
    this.setState((prevState, props) =>
      RangeType.select(month, RangeType.from(prevState))
    );
  };

  render() {
    const {fst, snd} = this.state;
    return (
      <Container>
        <Header>
          <H1>Select a range</H1>
        </Header>
        <Year year={2017} selectMonth={this.selectMonth} selection={{fst, snd}}>
          <Month position={1}>Jan</Month>
          <Month position={2}>Feb</Month>
          <Month position={3}>Mar</Month>
          <Month position={4}>Apr</Month>
          <Month position={5}>May</Month>
          <Month position={6}>Jun</Month>
          <Month position={7}>Jul</Month>
          <Month position={8}>Aug</Month>
          <Month position={9}>Sep</Month>
          <Month position={10}>Oct</Month>
          <Month position={11}>Nov</Month>
          <Month position={12}>Dec</Month>
        </Year>
        <Year year={2018} selectMonth={this.selectMonth} selection={{fst, snd}}>
          <Month position={1}>Jan</Month>
          <Month position={2}>Feb</Month>
          <Month position={3}>Mar</Month>
          <Month position={4}>Apr</Month>
          <Month position={5}>May</Month>
          <Month position={6}>Jun</Month>
          <Month position={7}>Jul</Month>
          <Month position={8}>Aug</Month>
          <Month position={9}>Sep</Month>
          <Month position={10}>Oct</Month>
          <Month position={11}>Nov</Month>
          <Month position={12}>Dec</Month>
        </Year>
        <Year year={2019} selectMonth={this.selectMonth} selection={{fst, snd}}>
          <Month position={1}>Jan</Month>
          <Month position={2}>Feb</Month>
          <Month position={3}>Mar</Month>
          <Month position={4}>Apr</Month>
          <Month position={5}>May</Month>
          <Month position={6}>Jun</Month>
          <Month position={7}>Jul</Month>
          <Month position={8}>Aug</Month>
          <Month position={9}>Sep</Month>
          <Month position={10}>Oct</Month>
          <Month position={11}>Nov</Month>
          <Month position={12}>Dec</Month>
        </Year>
      </Container>
    );
  }
}

export default MonthRange;
