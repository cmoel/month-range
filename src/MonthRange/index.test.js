import React from "react";
import renderer from "react-test-renderer";
import {mount} from "enzyme";

import MonthRange from "./index";
import MonthType from "./Month/MonthType";

test("renders correctly", () => {
  const component = renderer.create(<MonthRange />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  let twenty17 = tree.children[1].children[1];
  let apr2017 = twenty17.children[3];

  apr2017.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  twenty17 = tree.children[1].children[1];
  let feb2017 = twenty17.children[1];

  feb2017.props.onClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(component.getInstance().state).toEqual({
    fst: {year: 2017, position: 4},
    snd: {year: 2017, position: 2},
  });
});

describe("component state and events", () => {
  test("state.fst and state.snd are both undefined on initial render", () => {
    const component = mount(<MonthRange />);
    expect(component.state().fst).toBe(undefined);
    expect(component.state().snd).toBe(undefined);
    component.unmount();
  });

  test("clicking 1 month sets state.fst to that month", () => {
    const component = mount(<MonthRange />);
    const january = component.findWhere(
      c => c.props().position === 1 && c.props().year === 2018
    );
    january.simulate("click");

    expect(component.state().fst).toEqual(
      MonthType.from({position: 1, year: 2018})
    );
    expect(component.state().snd).toBe(undefined);
    component.unmount();
  });

  test("clicking 2 months sets state.fst to first month and state.snd to the second", () => {
    const component = mount(<MonthRange />);
    const january = component.findWhere(
      c => c.props().position === 1 && c.props().year === 2018
    );
    const june = component.findWhere(
      c => c.props().position === 6 && c.props().year === 2018
    );
    january.simulate("click");
    june.simulate("click");

    expect(component.state().fst).toEqual(
      MonthType.from({position: 1, year: 2018})
    );
    expect(component.state().snd).toEqual(
      MonthType.from({position: 6, year: 2018})
    );
    component.unmount();
  });
});
