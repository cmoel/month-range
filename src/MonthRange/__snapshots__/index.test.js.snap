// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.glamor-44,
[data-glamor-44] {
  margin: 42px auto 0;
  max-width: 320px;
}

.glamor-1,
[data-glamor-1] {
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.glamor-0,
[data-glamor-0] {
  color: #333;
  font-size: 1em;
  margin: 0;
  letter-spacing: 0.10em;
  text-transform: uppercase;
}

.glamor-2,
[data-glamor-2] {
  color: #333;
  font-size: 0.75em;
  font-weight: 400;
  margin-top: 2em;
  padding-bottom: 0.5em;
}

.glamor-15,
[data-glamor-15] {
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-flow: row wrap;
  padding: inherit;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-flow: row wrap;
}

.glamor-3,
[data-glamor-3] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: none;
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-8,
[data-glamor-8] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: 1px solid hsl(182, 11%, 71%);
  border-bottom: none;
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-9,
[data-glamor-9] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: 1px solid hsl(182, 11%, 71%);
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-14,
[data-glamor-14] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: 1px solid hsl(182, 11%, 71%);
  border-bottom: 1px solid hsl(182, 11%, 71%);
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

<div
  className="glamor-44"
>
  <header
    className="glamor-1"
  >
    <h1
      className="glamor-0"
    >
      Select a range
    </h1>
  </header>
  <article>
    <div
      className="glamor-2"
    >
      2017
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
  <article>
    <div
      className="glamor-2"
    >
      2018
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
  <article>
    <div
      className="glamor-2"
    >
      2019
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
</div>
`;

exports[`renders correctly 2`] = `
.glamor-44,
[data-glamor-44] {
  margin: 42px auto 0;
  max-width: 320px;
}

.glamor-1,
[data-glamor-1] {
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.glamor-0,
[data-glamor-0] {
  color: #333;
  font-size: 1em;
  margin: 0;
  letter-spacing: 0.10em;
  text-transform: uppercase;
}

.glamor-2,
[data-glamor-2] {
  color: #333;
  font-size: 0.75em;
  font-weight: 400;
  margin-top: 2em;
  padding-bottom: 0.5em;
}

.glamor-15,
[data-glamor-15] {
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-flow: row wrap;
  padding: inherit;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-flow: row wrap;
}

.glamor-3,
[data-glamor-3] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: none;
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-8,
[data-glamor-8] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: 1px solid hsl(182, 11%, 71%);
  border-bottom: none;
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-9,
[data-glamor-9] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: 1px solid hsl(182, 11%, 71%);
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-14,
[data-glamor-14] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: 1px solid hsl(182, 11%, 71%);
  border-bottom: 1px solid hsl(182, 11%, 71%);
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-6,
[data-glamor-6] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: none;
  background-color: hsl(221, 97%, 54%);
  color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

<div
  className="glamor-44"
>
  <header
    className="glamor-1"
  >
    <h1
      className="glamor-0"
    >
      Select a range
    </h1>
  </header>
  <article>
    <div
      className="glamor-2"
    >
      2017
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-6 fst"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
  <article>
    <div
      className="glamor-2"
    >
      2018
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
  <article>
    <div
      className="glamor-2"
    >
      2019
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
</div>
`;

exports[`renders correctly 3`] = `
.glamor-44,
[data-glamor-44] {
  margin: 42px auto 0;
  max-width: 320px;
}

.glamor-1,
[data-glamor-1] {
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.glamor-0,
[data-glamor-0] {
  color: #333;
  font-size: 1em;
  margin: 0;
  letter-spacing: 0.10em;
  text-transform: uppercase;
}

.glamor-2,
[data-glamor-2] {
  color: #333;
  font-size: 0.75em;
  font-weight: 400;
  margin-top: 2em;
  padding-bottom: 0.5em;
}

.glamor-15,
[data-glamor-15] {
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-flow: row wrap;
  padding: inherit;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-flow: row wrap;
}

.glamor-3,
[data-glamor-3] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: none;
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-8,
[data-glamor-8] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: 1px solid hsl(182, 11%, 71%);
  border-bottom: none;
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-9,
[data-glamor-9] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: 1px solid hsl(182, 11%, 71%);
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-14,
[data-glamor-14] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: 1px solid hsl(182, 11%, 71%);
  border-bottom: 1px solid hsl(182, 11%, 71%);
  background-color: #fff;
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-4,
[data-glamor-4] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: none;
  background-color: hsl(221, 97%, 54%);
  color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.glamor-5,
[data-glamor-5] {
  cursor: pointer;
  font-weight: 200;
  line-height: 2.5em;
  text-align: center;
  user-select: none;
  width: 52px;
  border-top: 1px solid hsl(182, 11%, 71%);
  border-left: 1px solid hsl(182, 11%, 71%);
  border-right: none;
  border-bottom: none;
  background-color: hsl(215, 50%, 94%);
  color: #333;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

<div
  className="glamor-44"
>
  <header
    className="glamor-1"
  >
    <h1
      className="glamor-0"
    >
      Select a range
    </h1>
  </header>
  <article>
    <div
      className="glamor-2"
    >
      2017
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-4 snd"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-5 selected"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-4 fst"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
  <article>
    <div
      className="glamor-2"
    >
      2018
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
  <article>
    <div
      className="glamor-2"
    >
      2019
    </div>
    <div
      className="glamor-15"
    >
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Jan
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Feb
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Mar
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        Apr
      </div>
      <div
        className="glamor-3"
        onClick={[Function]}
      >
        May
      </div>
      <div
        className="glamor-8"
        onClick={[Function]}
      >
        Jun
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Jul
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Aug
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Sep
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Oct
      </div>
      <div
        className="glamor-9"
        onClick={[Function]}
      >
        Nov
      </div>
      <div
        className="glamor-14"
        onClick={[Function]}
      >
        Dec
      </div>
    </div>
  </article>
</div>
`;
